pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/OfflineSegmentAssignment.java

```diff
@@ -133,6 +135,10 @@ private void checkReplication(InstancePartitions instancePartitions) {
   private List<String> assignSegment(String segmentName, Map<String, Map<String, String>> currentAssignment,
       InstancePartitions instancePartitions) {
     int numReplicaGroups = instancePartitions.getNumReplicaGroups();
+    if (_isDimTable) {
```

Comment:
can we add a new subclass of OfflineSegmentAssignment instead of using a boolean check?

Reply:
Thanks for the suggestion, this makes the code much cleaner and also allows to add an implementation of `rebalanceTable` method.